### PR TITLE
add login form

### PR DESCRIPTION
**Pull request by AbdallahGot**
add login fomr component and the function after submit to handle the login process

### CODE REVIEW

**Review by LinaYahya — CHANGES_REQUESTED** [review]

File: client/src/Components/Login/LoginFrom.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  const onFinish = async (values: loginProps) => {
+    try {
+      setLoading(true);
+      setTimeout(() => {
+        setLoading(false);
+      }, 1000);
+      const loginResult = await axios.post('/api/v1/login', values);
+      const newLink = `/user/${loginResult.data.data.providerName}`;
+      message.success(t('success-login'));
+      console.log(location);
```
Comment (LinaYahya) [inline client/src/Components/Login/LoginFrom.tsx:28]: remove these two lines,
Actually, we want to redirect user  after success so no need to display a message for user

Comment (LinaYahya) [inline client/src/Components/Login/LoginFrom.tsx:28]:
```suggestion
      
```

File: client/src/Components/Login/LoginFrom.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+  const [loading, setLoading] = useState(false);
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const location: any = useLocation();
+  const onFinish = async (values: loginProps) => {
+    try {
+      setLoading(true);
+      setTimeout(() => {
+        setLoading(false);
+      }, 1000);
```
Comment (LinaYahya) [inline client/src/Components/Login/LoginFrom.tsx:24]: No need,
Just after `axios`,  loginResult set it to false

File: client/src/Components/Login/LoginFrom.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      const loginResult = await axios.post('/api/v1/login', values);
+      const newLink = `/user/${loginResult.data.data.providerName}`;
+      message.success(t('success-login'));
+      console.log(location);
+      navigate(location.state?.from || newLink);
+    } catch (err: any) {
+      if (err.response) {
```
Comment (LinaYahya) [inline client/src/Components/Login/LoginFrom.tsx:31]:
```suggestion
             setLoading(false);
      if (err.response) {
```

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by RandSohail — APPROVED** [review]

**Review by abdulrahman-2020 — APPROVED** [review]

**Pull request merged**